### PR TITLE
Bittrex support 

### PR DESCRIPTION
**Pull request by imperator6**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Comment by anstak** [comment]
WOW! I have to waiting Bittrex so LONG... Why is it failed? Can you fixed it? It will be really cool!!!

**Comment by imperator6** [comment]
Hi, it's fixed now module node.bittrex.api was missing.

**Comment by askmike** [comment]
Thanks a lot! Great stuff!

Some notes:

- [x] Could you remove the README additions? Documentation should go in the docs folder, which is rendered here: https://gekko.wizb.it/docs/installation/installing_gekko.html
- [x] https://github.com/askmike/gekko/pull/968/files#diff-7c8f7b45a5eee77a5176e4b7e7de8c83R214 do you need this even after setting the minimum amount in `getCapabilities `?
- [ ] https://github.com/askmike/gekko/pull/968/files#diff-da55d303c7e6c76807d65bddbfbbcdcdR50 what is that doing there?
- [x] the tests you added are not unit tests right? Do they interfere with npm test?
- [x] https://github.com/askmike/gekko/pull/968/files#diff-da55d303c7e6c76807d65bddbfbbcdcdR300 that is very specific functionality we just need for bittrex I think?

**Comment by imperator6** [comment]
https://github.com/askmike/gekko/pull/968/files#diff-7c8f7b45a5eee77a5176e4b7e7de8c83R214
 If we would't catch the DUST_TRADE_DISALLOWED_MIN_VALUE_50K_SAT it will end up in an endless loop here as the order enters the recursive retry. The `getCapabilities` has no effect on this behavior as the issue seems to be timing related in the cancel and place new order process. On bittrex it seems to be that the portfolio amount is incorrect for a while after an order has been canceled. 

1. place a buy order -> currency amount in the wallet is no something close to zero
2. cancel the order 
3. instantly place the new order (calling `getPortfolio`) will still result in something close zero and execute the order with the incorrect very small amount

Thats the reason why I've added the 10sec sleep bevore replacing the order. 
https://github.com/askmike/gekko/pull/968/files#diff-da55d303c7e6c76807d65bddbfbbcdcdR300 

New `keepAsset `function 
https://github.com/askmike/gekko/pull/968/commits/516f9b24c49f303a3a65e1d06833189eb6239cf7#diff-da55d303c7e6c76807d65bddbfbbcdcd

if you put a 'keepAsset' in the config, gekko wil not sell all your assets. :-)

```
config.watch = {

  exchange: 'bittrex',
  currency: 'USDT',
  asset: 'BCC',
  keepAsset: 50,
}
```

**Comment by askmike** [comment]
> Thats the reason why I've added the 10sec sleep bevore replacing the order.

Great, but maybe we can only wait 10 seconds on bittrex (and not on other exchanges, by checking whether this exchange is bittrex), what do you think?

> if you put a 'keepAsset' in the config, gekko wil not sell all your assets. :-)

Great idea! There is a discussion ongoing about different ways to handle this, see #937. Would you mind if we don't include this proposal just yet inside this PR? I've noted your solution in that issue :)

Would be great to get bittrex in there, thanks a lot for your work and testing!

**Comment by grazianogrespan** [comment]
Hello all! thanks @imperator6 for your PR, it's great! I've started developing Bittrex integration few days ago.. and you beat me to the punch! :sweat_smile:
Just a question, is there a reason why you chose to put only 4 pairs on `getCapabilities()` markets?
Bittrex supports many different assets and pairs.

**Comment by askmike** [comment]
The last thing we need is:

> Thats the reason why I've added the 10sec sleep bevore replacing the order.

Only apply the 10 second wait on bittrex, not other exchanges. I can also pull in this PR and fix that myself? Because I am already very happy with your awesome work :)

**Comment by imperator6** [comment]
@grazianogrespan no there is no reason. I've just started testing with them, but I was to lazy to add the rest. But it should be easy to add the rest. Currently I'm bussy on my real work :-). I've you like I can give you acces to my fork of gekko. https://github.com/imperator6/gekko so that you can add them.

**Comment by imperator6** [comment]
@askmike I wil fix the 10sec wait on bittrex soon as poossible.

**Comment by askmike** [comment]
@imperator6 thanks a lot! There is no rush :)

**Comment by askmike** [comment]
Awesome stuff! You fixed everything on my list, can I merge?

**Comment by khaledabbas** [comment]
I pulled this PR locally but still can't see bittrex appear in any of the exchange dropdown lists through the app

**Comment by imperator6** [comment]
Try npm install before starting

**Comment by khaledabbas** [comment]
Yup, did the trick! thanks @imperator6!

**Comment by khaledabbas** [comment]
It appeared now in live gekko only though. Still not there in importer.

**Comment by khaledabbas** [comment]
@imperator6 Would it be hard to add Bittrex to importer exchanges as well (`importers/exchanges`)? Thanks for the great effort 👍

**Comment by imperator6** [comment]
@khaledabbas Bittrex only returns the last ~500 trades as market history. I found no way to get more. Just start a watcher and run it for a few days :-)

@askmike I would recomment to wait a few days. I still doing some backtesting with the lastest code. As more people are joining this branch it also a good opportunity to get some feedback.

File: exchanges/bittrex.js [hunk 1]
```diff
@@ -351,34 +351,312 @@ Trader.prototype.getTrades = function(since, callback, descending) {
...
+      'CVC', 'DAR', 'DASH', 'DCR', 'DCT', 'DGB', 'DGC', 'DGD', 'DMD', 'DOGE',
+      'DOPE', 'DRACO', 'DTB', 'DTC', 'DYN', 'EBST', 'EDG', 'EFL', 'EGC', 'EMC',
+      'EMC2', 'ENRG', 'ERC', 'ETC', 'ETH', 'EXCL', 'EXP', 'FAIR', 'FC2', 'FCT',
+      'FLDC', 'FLO', 'FRK', 'FSC2', 'FTC', 'FUN', 'GAM', 'GAME', 'GBG', 'GBYTE',
+      'GCR', 'GEMZ', 'GEO', 'GHC', 'GLD', 'GNO', 'GNT', 'GOLOS', 'GP', 'GRC',
+      'GRS', 'GRT', 'GUP', 'HKG', 'HMQ', 'HYPER', 'HZ', 'INCNT', 'INFX', 'IOC',
+      'ION', 'IOP', 'J', 'JWL', 'KMD', 'KORE', 'KR', 'LBC', 'LGD', 'LMC', 'LSK',
+      'LTC', 'LUN', 'LXC', 'MAID', 'MAX', 'MCO', 'MEC', 'MEME', 'METAL', 'MLN',
+      'MND', 'MONA', 'MTL', 'MTR', 'MUE', 'MUSIC', 'MYR', 'MYST', 'MZC', 'NAUT',
+      'NAV', 'NBT', 'NEOS', 'NET', 'NEU', 'NLG', 'NMR', 'NTRN', 'NXC', 'NXS',
```
Comment (mluisbrown) [inline exchanges/bittrex.js]: NEOS should be NEO. Was just trying out BTC/NEO market watcher on Bittrex. So glad you made this PR, as I was about to do it myself!

**Comment by mluisbrown** [comment]
Just an FYI that "NEOS" in the list of Bittrex assets should be "NEO" (made the same comment on the commit too, sorry).

**Comment by askmike** [comment]
@imperator6 merging this as I want to extend the timeout (between cancel & retry) func for gdax (see #999)

**Comment by nethergrim** [comment]
Hey guys, I can see an error on Bittrex:

```
2017-09-23T20:03:41.637141+00:00 app[worker.1]: 2017-09-23 20:03:41 (INFO):	We have new trading advice!
2017-09-23T20:03:41.637217+00:00 app[worker.1]: 2017-09-23 20:03:41 (INFO):		 Position: short
2017-09-23T20:03:41.637301+00:00 app[worker.1]: 2017-09-23 20:03:41 (INFO):		 Market price: 0.00028837
2017-09-23T20:03:41.637398+00:00 app[worker.1]: 2017-09-23 20:03:41 (INFO):		 Based on market time: 2017-09-23 20:02:00
2017-09-23T20:03:41.637451+00:00 app[worker.1]: 
2017-09-23T20:03:41.637638+00:00 app[worker.1]: 2017-09-23 20:03:41 (INFO):	Trader Received advice to go short. Selling  SWIFT
2017-09-23T20:03:41.638066+00:00 app[worker.1]: 2017-09-23 20:03:41 (DEBUG):	getTicker called
2017-09-23T20:03:41.713997+00:00 app[worker.1]: 2017-09-23 20:03:41 (DEBUG):	getTicker result { Bid: 0.00028837, Ask: 0.00028841, Last: 0.00028841 }
2017-09-23T20:03:41.714244+00:00 app[worker.1]: 2017-09-23 20:03:41 (DEBUG):	getPortfolio called
2017-09-23T20:03:41.789806+00:00 app[worker.1]: 2017-09-23 20:03:41 (DEBUG):	getPortfolio result: [ { name: 'SWIFT', amount: 0 },
2017-09-23T20:03:41.789814+00:00 app[worker.1]:   { name: 'BTC', amount: 0.03625238 } ]
2017-09-23T20:03:41.790032+00:00 app[worker.1]: 2017-09-23 20:03:41 (DEBUG):	getFee called
2017-09-23T20:03:41.790999+00:00 app[worker.1]: 2017-09-23 20:03:41 (ERROR):	Wanted to buy BTC but the amount is too small (0.000000000000) at Bittrex
```

**Comment by John1231983** [comment]
Thanks for support bittrex. Could you provide the function to import data from UI, and perform backtest? Thanks. I did not find any import bittrex from UI

**Comment by flux154** [comment]
@askmike  firstly, thanks for creating gekko and sharing the knowledge.

I have run a number of backtests and arrived at a formula I am happy with, I connected to bittrex and left it running all night. When I came to it this morning there were no trades, so I stopped it.
I went to the backtest page and the bittrex data was there, so I ran a backtest and it showed I should have had 7 trades.
screen shot 2018-01-26 at 09 10 35


Unsure where to go from here.

Edit: I can see gekko in my whitelist on bittrex from last night, but connecting now and I can't see it.

Edit2: I realised I was getting BUY_IP_VIOLATION, I entered the wrong IP address last night, when I entered the correct one I forgot to delete the other, just waiting to see if this allows gekko to start trading, ui looking good.

Edit3: Not sure if gekko is getting out of paper trading mode. The summary shows I am connecting, gekko is receiving candle data but not making any buy/sells, despite me selecting livebot.

**Pull request merged**